src/aio.rs

```diff
@@ -577,3 +577,91 @@ impl ConnectionLike for MultiplexedConnection {
         self.db
     }
 }
+
+#[cfg(feature = "tokio-rt-core")]
+mod connection_manager {
+    use super::*;
+
+    /// A `ConnectionManager` is a proxy that wraps a multiplexed connection and
+    /// automatically reconnects to the server when necessary.
+    pub struct ConnectionManager {
+        connection_info: ConnectionInfo,
+        connection: MultiplexedConnection,
+    }
+
+    #[cfg(feature = "tokio-rt-core")]
+    impl ConnectionManager {
+        /// Connect to the server and store the connection inside the returned `ConnectionManager`.
+        ///
+        /// TODO: Add `new_tokio`?
+        pub async fn new(connection_info: ConnectionInfo) -> RedisResult<Self> {
+            let (connection, driver) = MultiplexedConnection::new(connect(&connection_info).await?);
+            tokio::spawn(driver);
+            Ok(Self {
+                connection_info,
+                connection,
+            })
+        }
+
+        async fn reconnect(&mut self) -> RedisResult<&mut Self> {
+            let (new_connection, driver) = MultiplexedConnection::new(connect(&self.connection_info).await?);
+            tokio::spawn(driver);
+            let _old_connection = std::mem::replace(&mut self.connection, new_connection);
+            Ok(self)
```

Comment:
This seems quite straightforward. The old connection will be dropped, resulting in cleanup. That should work, right?